### PR TITLE
[New Rule] Google Workspace - Suspended User Account Renewed

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/security-team/issues/5915

## Summary
The rule detects when a suspended user account is renewed in Google Workspace. While this activity is expected in environments, it should be rather low volume or require further investigation by analysts. 

Typically a user may be suspended due to employee termination in which administrators would then transfer drive contents to another existing user before ultimately deleting the user and removing their Google Workspace license. Therefore, renewals should be low volume.

To accomplish this, an adversary would need a compromised administrator account or have domain-wide delegation setup for a service account in GCP where the scope of OAuth includes admin privileges to the Directory API. Another option is to abuse container-bound scripts where a phishing email is sent to an administrator and code execution occurs on behalf of the admin to accomplish this.

Assuming this rule is not low volume as expected, there is not much room for tuning based on current datasets, therefore deprecation would need to occur. 

<details>
<summary>Example Data</summary>

```sql
{
  "_index": ".ds-logs-google_workspace.admin-default-2022.12.22-000006",
  "_id": "PB/jX8uj7SAwjbuc74pwVcWtBB4=",
  "_version": 1,
  "_score": 0,
  "_source": {
    "agent": {
      "name": "ubuntu-server-tdejesus",
      "id": "f4011165-a1b8-4d8e-b902-56ba8835cc28",
      "ephemeral_id": "41da8e24-c235-4d0f-87b8-760b1feffd7c",
      "type": "filebeat",
      "version": "8.5.0"
    },
    "elastic_agent": {
      "id": "f4011165-a1b8-4d8e-b902-56ba8835cc28",
      "version": "8.5.0",
      "snapshot": false
    },
    "source": {
      "geo": {
        "continent_name": "North America",
        "region_iso_code": "US-OH",
        "city_name": "North Canton",
        "country_iso_code": "US",
        "country_name": "United States",
        "region_name": "Ohio",
        "location": {
          "lon": -81.3798,
          "lat": 40.7961
        }
      },
      "as": {
        "number": 10796,
        "organization": {
          "name": "TWC-10796-MIDWEST"
        }
      },
      "ip": "2603:6010:ed00:b789:d047:3a3a:cbf4:736b",
      "user": {
        "domain": "dejesusarcheology.com",
        "name": "terrance",
        "id": "115903088752625509360",
        "email": "terrance@dejesusarcheology.com"
      }
    },
    "tags": [
      "forwarded",
      "google-workspace-admin"
    ],
    "cloud": {
      "availability_zone": "us-east1-b",
      "instance": {
        "name": "ubuntu-server-tdejesus",
        "id": "1709224677170316971"
      },
      "provider": "gcp",
      "service": {
        "name": "GCE"
      },
      "machine": {
        "type": "e2-medium"
      },
      "project": {
        "id": "elastic-security-dev"
      },
      "account": {
        "id": "elastic-security-dev"
      }
    },
    "input": {
      "type": "httpjson"
    },
    "@timestamp": "2023-01-17T02:11:53.573Z",
    "ecs": {
      "version": "8.5.0"
    },
    "related": {
      "ip": [
        "2603:6010:ed00:b789:d047:3a3a:cbf4:736b"
      ],
      "user": [
        "terrance",
        "dennis"
      ]
    },
    "google_workspace": {
      "actor": {
        "type": "USER"
      },
      "kind": "admin#reports#activity",
      "admin": {
        "user": {
          "email": "dennis@dejesusarcheology.com"
        }
      },
      "event": {
        "type": "USER_SETTINGS"
      }
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "google_workspace.admin"
    },
    "organization": {
      "id": "C00qtspd5"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2023-01-17T02:16:47Z",
      "provider": "admin",
      "created": "2023-01-17T02:16:46.574Z",
      "kind": "event",
      "action": "UNSUSPEND_USER",
      "id": "-8792325838351234245",
      "category": [
        "iam"
      ],
      "type": [
        "change",
        "user"
      ],
      "dataset": "google_workspace.admin"
    },
    "user": {
      "domain": "dejesusarcheology.com",
      "name": "terrance",
      "id": "115903088752625509360",
      "email": "terrance@dejesusarcheology.com",
      "target": {
        "domain": "dejesusarcheology.com",
        "name": "dennis",
        "email": "dennis@dejesusarcheology.com"
      }
    }
  },
  "fields": {
    "elastic_agent.version": [
      "8.5.0"
    ],
    "event.category": [
      "iam"
    ],
    "source.user.email": [
      "terrance@dejesusarcheology.com"
    ],
    "cloud.availability_zone": [
      "us-east1-b"
    ],
    "source.user.name.text": [
      "terrance"
    ],
    "user.target.email": [
      "dennis@dejesusarcheology.com"
    ],
    "source.geo.region_name": [
      "Ohio"
    ],
    "google_workspace.actor.type": [
      "USER"
    ],
    "source.ip": [
      "2603:6010:ed00:b789:d047:3a3a:cbf4:736b"
    ],
    "agent.name": [
      "ubuntu-server-tdejesus"
    ],
    "user.target.name.text": [
      "dennis"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "source.geo.region_iso_code": [
      "US-OH"
    ],
    "event.kind": [
      "event"
    ],
    "google_workspace.event.type": [
      "USER_SETTINGS"
    ],
    "source.geo.city_name": [
      "North Canton"
    ],
    "user.id": [
      "115903088752625509360"
    ],
    "input.type": [
      "httpjson"
    ],
    "data_stream.type": [
      "logs"
    ],
    "related.user": [
      "terrance",
      "dennis"
    ],
    "user.target.name": [
      "dennis"
    ],
    "tags": [
      "forwarded",
      "google-workspace-admin"
    ],
    "cloud.machine.type": [
      "e2-medium"
    ],
    "cloud.provider": [
      "gcp"
    ],
    "event.provider": [
      "admin"
    ],
    "cloud.service.name": [
      "GCE"
    ],
    "agent.id": [
      "f4011165-a1b8-4d8e-b902-56ba8835cc28"
    ],
    "ecs.version": [
      "8.5.0"
    ],
    "event.created": [
      "2023-01-17T02:16:46.574Z"
    ],
    "google_workspace.admin.user.email": [
      "dennis@dejesusarcheology.com"
    ],
    "organization.id": [
      "C00qtspd5"
    ],
    "agent.version": [
      "8.5.0"
    ],
    "source.user.name": [
      "terrance"
    ],
    "source.as.number": [
      10796
    ],
    "user.name": [
      "terrance"
    ],
    "source.geo.location": [
      {
        "coordinates": [
          -81.3798,
          40.7961
        ],
        "type": "Point"
      }
    ],
    "cloud.instance.id": [
      "1709224677170316971"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "google_workspace"
    ],
    "user.email": [
      "terrance@dejesusarcheology.com"
    ],
    "related.ip": [
      "2603:6010:ed00:b789:d047:3a3a:cbf4:736b"
    ],
    "source.geo.country_iso_code": [
      "US"
    ],
    "user.target.domain": [
      "dejesusarcheology.com"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "source.user.id": [
      "115903088752625509360"
    ],
    "user.domain": [
      "dejesusarcheology.com"
    ],
    "source.as.organization.name.text": [
      "TWC-10796-MIDWEST"
    ],
    "elastic_agent.id": [
      "f4011165-a1b8-4d8e-b902-56ba8835cc28"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "source.as.organization.name": [
      "TWC-10796-MIDWEST"
    ],
    "source.geo.continent_name": [
      "North America"
    ],
    "google_workspace.kind": [
      "admin#reports#activity"
    ],
    "event.ingested": [
      "2023-01-17T02:16:47.000Z"
    ],
    "event.action": [
      "UNSUSPEND_USER"
    ],
    "@timestamp": [
      "2023-01-17T02:11:53.573Z"
    ],
    "cloud.account.id": [
      "elastic-security-dev"
    ],
    "event.type": [
      "change",
      "user"
    ],
    "data_stream.dataset": [
      "google_workspace.admin"
    ],
    "agent.ephemeral_id": [
      "41da8e24-c235-4d0f-87b8-760b1feffd7c"
    ],
    "source.user.domain": [
      "dejesusarcheology.com"
    ],
    "event.id": [
      "-8792325838351234245"
    ],
    "source.geo.country_name": [
      "United States"
    ],
    "event.dataset": [
      "google_workspace.admin"
    ],
    "cloud.instance.name": [
      "ubuntu-server-tdejesus"
    ],
    "user.name.text": [
      "terrance"
    ],
    "cloud.project.id": [
      "elastic-security-dev"
    ]
  }
}
```

</details>

<details>
<summary>Screenshots</summary>

Screenshot 2023-02-22 at 2 12 18 PM

</details>

### CODE REVIEW

File: rules/integrations/google_workspace/initial_access_google_workspace_suspended_user_renewed.toml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+references = ["https://support.google.com/a/answer/1110339"]
+risk_score = 21
+rule_id = "00678712-b2df-11ed-afe9-f661ea17fbcc"
+severity = "low"
+tags = ["Elastic", "Cloud", "Google Workspace", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset: google_workspace.admin and event.category: iam and event.action: UNSUSPEND_USER
```
Comment (brokensound77) [inline rules/integrations/google_workspace/initial_access_google_workspace_suspended_user_renewed.toml:50]:
```suggestion
event.dataset:google_workspace.admin and event.category:iam and event.action:UNSUSPEND_USER
```

**Review by imays11 — APPROVED** [review]
Looks good 👍🏾

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Review by brokensound77 — APPROVED** [review]
LGTM

**Pull request merged**